### PR TITLE
Add support for arm64

### PR DESCRIPTION
**Pull request by galak**
Add ability to specify ARCH=arm64 to get a deployment for arm64

### CODE REVIEW

**Comment by pda** [comment]
Fantastic!
I'll need to take a closer look in coming days; it might not be until later in the week.
I'm keen to see some Gravitron2 buildkite-agent elastic stacks running :)

**Comment by galak** [comment]
https://github.com/buildkite/agent/issues/1228

> Fantastic!
> I'll need to take a closer look in coming days; it might not be until later in the week.
> I'm keen to see some Gravitron2 buildkite-agent elastic stacks running :)

Cool, there are few rough edges (Like getting https://github.com/buildkite/agent/issues/1228 fixed) and figuring out what todo about docker compose on arm64.  But figured it was worth sharing as is for discussion.

File: packer/linux/conf/bin/bk-install-elastic-stack.sh [hunk 1]
```diff
@@ -62,13 +62,13 @@ export PLUGINS_ENABLED="${PLUGINS_ENABLED[*]-}"
...
-	chmod +x /usr/bin/buildkite-agent-edge
-	buildkite-agent-edge --version
-fi
+#if [[ "${BUILDKITE_AGENT_RELEASE}" == "edge" ]] ; then
+#	echo "Downloading buildkite-agent edge..."
+#	curl -Lsf -o /usr/bin/buildkite-agent-edge \
+#		"https://download.buildkite.com/agent/experimental/latest/buildkite-agent-linux-amd64"
+#	chmod +x /usr/bin/buildkite-agent-edge
+#	buildkite-agent-edge --version
+#fi
```
Comment (nitrocode) [inline packer/linux/conf/bin/bk-install-elastic-stack.sh:71]: why was this commented out? Shoudn't this be left here and match the architecture ?

File: packer/linux/scripts/install-buildkite-agent.sh [hunk 2]
```diff
@@ -12,16 +20,19 @@ sudo useradd --base-dir /var/lib --uid 2000 buildkite-agent
...
-echo "Downloading buildkite-agent beta..."
-sudo curl -Lsf -o /usr/bin/buildkite-agent-beta \
-  "https://download.buildkite.com/agent/unstable/latest/buildkite-agent-linux-amd64"
-sudo chmod +x /usr/bin/buildkite-agent-beta
-buildkite-agent-beta --version
+#echo "Downloading buildkite-agent beta..."
+#sudo curl -Lsf -o /usr/bin/buildkite-agent-beta \
+#  "https://download.buildkite.com/agent/unstable/latest/buildkite-agent-linux-arm64"
+#sudo chmod +x /usr/bin/buildkite-agent-beta
+#buildkite-agent-beta --version
```
Comment (nitrocode) [inline packer/linux/scripts/install-buildkite-agent.sh:35]: why was this commented out? Shoudn't this be left here and match the architecture of the machine ?

File: packer/linux/scripts/install-docker.sh [hunk 3]
```diff
@@ -30,18 +31,18 @@ sudo curl -Lfs -o /etc/systemd/system/docker.socket https://raw.githubuserconten
...
-echo "Downloading docker-compose..."
-sudo curl -Lsf -o /usr/bin/docker-compose https://github.com/docker/compose/releases/download/${DOCKER_COMPOSE_VERSION}/docker-compose-Linux-x86_64
-sudo chmod +x /usr/bin/docker-compose
-docker-compose --version
+if [ ${MACHINE} == "x86_64" ]; then
+	echo "Downloading docker-compose..."
+	sudo curl -Lsf -o /usr/bin/docker-compose https://github.com/docker/compose/releases/download/${DOCKER_COMPOSE_VERSION}/docker-compose-Linux-x86_64
+	sudo chmod +x /usr/bin/docker-compose
+	docker-compose --version
+fi
```
Comment (nitrocode) [inline packer/linux/scripts/install-docker.sh:39]:
```suggestion
else
  sudo pip3 install docker-compose
fi
```

Wouldn't this be enough to install docker-compose directly on arm?

**Comment by nitrocode** [comment]
@galak appreciate the PR for this as ARM is definitely more cost effective. If you have time, do you mind going over the PR comments and merging with master?

**Comment by galak** [comment]
> @galak appreciate the PR for this as ARM is definitely more cost effective. If you have time, do you mind going over the PR comments and merging with master?

I'm happy to make the updates and look over the comments.  As far as merging the PR, we'd need someone at buildkite to do that.

File: packer/linux/conf/bin/bk-install-elastic-stack.sh (hunk 1 above)
Comment (galak) [reply packer/linux/conf/bin/bk-install-elastic-stack.sh:71]: At the time I pushed the PR there was an issue with the prebuild packages not existing for aarch64.  This is fixed now so I'll rework this.

**Comment by nitrocode** [comment]
> I'm happy to make the updates and look over the comments.

Thank you

> As far as merging the PR, we'd need someone at buildkite to do that.

Of course. I'm sure once the fixes are in place, we can give a friendly @ to chloeruka or pda, juanitofatos, sj26.

**Comment by nitrocode** [comment]
@galak friendly ping. Any time table on fixing the conflicts ? Thanks again for your work.

**Comment by yob** [comment]
Thanks for getting the ball rolling on this @galak, we've just merged a PR (#758, based on this one) that adds support for ARM instance types.

**Pull request closed without merging**